internal/pkg/cli/run_local.go

```diff
@@ -409,6 +420,116 @@ func (o *runLocalOpts) getTask(ctx context.Context) (orchestrator.Task, error) {
 	return task, nil
 }
 
+func (o *runLocalOpts) prepareTask(ctx context.Context) (orchestrator.Task, error) {
+	task, err := o.getTask(ctx)
+	if err != nil {
+		return orchestrator.Task{}, fmt.Errorf("get task: %w", err)
+	}
+
+	mft, _, err := workloadManifest(&workloadManifestInput{
+		name:         o.wkldName,
+		appName:      o.appName,
+		envName:      o.envName,
+		ws:           o.ws,
+		interpolator: o.newInterpolator(o.appName, o.envName),
+		unmarshal:    o.unmarshal,
+		sess:         o.envManagerSess,
+	})
+	if err != nil {
+		return orchestrator.Task{}, err
+	}
+
+	containerURIs, err := o.buildContainerImages(mft)
+	if err != nil {
+		return orchestrator.Task{}, fmt.Errorf("build images: %w", err)
+	}
+
+	// replace built images with the local built URI
+	for name, uri := range containerURIs {
+		ctr, ok := task.Containers[name]
+		if !ok {
+			return orchestrator.Task{}, fmt.Errorf("built an image for %q, which doesn't exist in the task", name)
+		}
+
+		ctr.ImageURI = uri
+		task.Containers[name] = ctr
+	}
+
+	return task, nil
+}
+
+func (o *runLocalOpts) watchLocalFiles(stopCh <-chan struct{}) (<-chan interface{}, <-chan error, error) {
+	copilotDir := o.ws.Path()
+
+	watchCh := make(chan interface{})
+	watchErrCh := make(chan error)
+
+	watcher, err := o.newRecursiveWatcher()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if err = watcher.Add(copilotDir); err != nil {
+		return nil, nil, err
+	}
+
+	watcherEvents := watcher.Events()
+	watcherErrors := watcher.Errors()
+
+	debounceTimer := time.NewTimer(0)
+	<-debounceTimer.C // flush channel to create timer for later use with Reset()
+
+	go func() {
+		for {
+			select {
+			case <-stopCh:
+				watcher.Close()
```

Comment:
Because this function needs to return the channels for events and errors while the watcher is still running. If we do `defer watcher.Close()` it'll close the watcher as the channels are returned, and so we won't get any information from them.